frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,202 @@ import {
+  const [dataChanged, setDataChanged] = React.useState<0 | 1>(0);
 
-  const tableData = React.useMemo(() => {
-    if (!search) return donors;
-    return donors.filter(
-      (donor) =>
-        donor.businessName.toLowerCase().includes(search.toLowerCase()) ||
-        donor.email.toLowerCase().includes(search.toLowerCase()) ||
-        donor.firstName
-          .toLowerCase()
-          .concat(" ", donor.lastName.toLowerCase())
-          .includes(search.toLowerCase()) ||
-        donor.phoneNumber.toLowerCase().includes(search.toLowerCase()),
-    );
-  }, [search, donors]);
+  const filterOptions = [
+    {
+      value: "all",
+      label: "All accounts",
+    },
+    {
+      value: "volunteers",
+      label: "Volunteers",
+    },
+    {
+      value: "donors",
+      label: "Donors",
+    },
+  ];
+
+  // Merge donors and volunteers into one list of type UserMgmtTableRecord
+  const buildTableUsers = (): UserMgmtTableRecord[] => {
+    const mergedUsers: UserMgmtTableRecord[] = [];
+
+    volunteers.forEach(function (volunteer) {
+      mergedUsers.push({
+        userId: volunteer.userId,
+        id: volunteer.id,
+        pointOfContact: `${volunteer.firstName} ${volunteer.lastName}`,
+        company: "",
+        email: volunteer.email,
+        phoneNumber: volunteer.phoneNumber,
+        accountType: volunteer.role,
+        approvalStatus: volunteer.status,
+      });
+    });
+
+    donors.forEach(function (donor) {
+      mergedUsers.push({
+        userId: donor.userId,
+        id: donor.id,
+        pointOfContact: `${donor.firstName} ${donor.lastName}`,
+        company: donor.businessName,
+        email: donor.email,
+        phoneNumber: donor.phoneNumber,
+        accountType: donor.role,
+        approvalStatus: "",
+      });
+    });
+
+    return mergedUsers;
+  };
 
   React.useEffect(() => {
     const getDonors = async () => {
       const res = await DonorAPIClient.getAllDonors();
       setDonors(res);
     };
 
+    const getVolunteers = async () => {
+      const res = await VolunteerAPIClient.getAllVolunteers();
+      setVolunteers(res);
+    };
+
     getDonors();
-  }, []);
+    getVolunteers();
+  }, [dataChanged]);
+
+  // Update users whenever donors or volunteers have a change
+  React.useEffect(() => {
+    setUsers(buildTableUsers());
+  }, [donors, volunteers]);
+
+  const tableData = React.useMemo(() => {
+    let filteredUsers;
+    if (selectedFilter === "all") filteredUsers = users;
+    else if (selectedFilter === "volunteers")
+      filteredUsers = users.filter(
+        (user) => user.accountType === Role.VOLUNTEER,
+      );
+    else
+      filteredUsers = users.filter((user) => user.accountType === Role.DONOR);
+
+    if (!search) return filteredUsers;
+    return users.filter(
+      (user) =>
+        user.pointOfContact.toLowerCase().includes(search.toLowerCase()) ||
+        user.company.toLowerCase().includes(search.toLowerCase()) ||
```

Comment:
```suggestion
      (user) =>
        user.accountType.toLowerCase().includes(search.toLowerCase()) ||
        user.pointOfContact.toLowerCase().includes(search.toLowerCase()) ||
        user.company.toLowerCase().includes(search.toLowerCase()) ||
```
because:
No search/filter state:
![image](https://user-images.githubusercontent.com/19617248/159202055-08887000-866a-40a9-913c-7e6672272885.png)
"volunteer" entered in search: (showing 2 results instead of 3)
![image](https://user-images.githubusercontent.com/19617248/159202080-dd6883b4-008d-43d1-b30b-ccf892b0409f.png)